File: quartz-energy/src/backend/QuartzEnergy.Common.Services/Services/Statistics/NoRequestStatisticsService.cs
namespace QuartzEnergy.Common.Services.Services.Statistics
{
    using QuartzEnergy.Common.Dal.Infrastructure;
    using QuartzEnergy.Common.Services.Models.Statistics;

    public abstract class NoRequestStatisticsService<TData>
        : StatisticsService<EmptyStatisticsRequest, TData>
    {
        protected NoRequestStatisticsService(IUnitOfWorkFactory unitOfWorkFactory)
            : base(unitOfWorkFactory)
        {
        }
    }
}

File: quartz-energy/src/frontend/src/app/common/services/statistics/concrete/no-request-statistics.service.ts
import { EmptyStatisticsRequest } from './../../../models/statistics/empty-statistics-request';
import { Injectable } from '@angular/core';
import { StatisticsService } from './statistics.service';

@Injectable()
export abstract class NoRequestStatisticsService<TData>
  extends StatisticsService<EmptyStatisticsRequest, TData> {
}
